### PR TITLE
CLN: cleaning core/common.py

### PR DESCRIPTION
**Pull request by jreback**
partial on #12503
- create `pandas/types`
- moved _some_ dtype to `pandas/types`
- move _some_ missing value utils to `pandas/core/missing`
- move `.take` and `.diff` to `core/algorithms`
- create `formats` with `core/formats,style` moved and `printing.py` (form `core/common.py`)
- removed some non-used functions

### CODE REVIEW

**Comment by jreback** [comment]
cc @wesm 

still more to do, but this is a start.

**Comment by jreback** [comment]
@shoyer @jorisvandenbossche @sinhrks @TomAugspurger 

any comments. don't want to leave this out too long as it touches lots of things.

**Comment by TomAugspurger** [comment]
Not objections to your description.

Not a big deal, but does `formatting/` make more sense than `formats/`?

**Comment by jreback** [comment]
its slightly longer, but no big deal.

**Comment by shoyer** [comment]
assuming you're just moving stuff around, this seems sane to me

**Comment by wesm** [comment]
I'll look later this AM (pacific time)

File: pandas/types/tests/test_generic.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import nose
+import numpy as np
+import pandas as pd
+import pandas.core.common as com
+import pandas.util.testing as tm
+
+_multiprocess_can_split_ = True
+
+
+class TestABCClasses(tm.TestCase):
```
Comment (wesm) [inline pandas/types/tests/test_generic.py]: Put these tests instead in `pandas/tests/types`?

**Comment by wesm** [comment]
This all looks fine to me. 

Aside: I'm on vacation most of this month but planning to spend some time in May / June / July on refactoring to look toward retooling DataFrame internals (it would be nice to plot a course to removing the BlockManager, deprecating Panel, and some of the other under-the-hood internals work we've been discussing). In the meantime, anything that can be done to get organized and compartmentalize non-user code that will be affected will help this proceed more quickly. I think this patch helps us along this path.

File: pandas/types/tests/test_generic.py (hunk 1 above)
Comment (jreback) [inline pandas/types/tests/test_generic.py]: done

**Comment by jreback** [comment]
np on schedule. will be a bit more internal refactoring soon.

**Comment by jreback** [comment]
@TomAugspurger do you feel strongly on the ``formats`` -> ``formatting``?

**Comment by TomAugspurger** [comment]
Not at all, especially since this is all internal and not user facing.

**Comment by gfyoung** [comment]
@jreback : Are you missing something in `setup.py`?  When I try installing the `master` branch, the `formats` folder does not get installed, leading to the error "No module named formats.printing"

**Comment by jreback** [comment]
I pushed a change.

though I installed fine from a fresh clone (this had to do with package setup)

**Comment by jreback** [comment]
@gfyoung you prob got somethign like this: https://ci.appveyor.com/project/jreback/pandas-465/build/1.0.409/job/fnp6uhksqcp039g4

**Comment by gfyoung** [comment]
@jreback : Indeed, I did, and I'm happy to say that with your most recent commit, everything installs just fine now!  Thanks!

**Comment by jreback** [comment]
great not sure why our install test on Travis didn't fail 
but I guess it's not actually testing the built version but just creating it

in any event appveyor catches it because of the way it builds (it's a full conda build) not just an inplace build

**Comment by jreback** [comment]
@TomAugspurger I think the doc build broke: lots of Styler warnings now. https://travis-ci.org/pydata/pandas/jobs/121316764.

building locally so far so good though (not done yet)

**Comment by jorisvandenbossche** [comment]
Yes, just wanted to mention the same. It's the same error again as before. I can't remember anymore what solved it before

**Comment by TomAugspurger** [comment]
Mmm, I think it magically fixed itself last time. Will look later.

**Comment by jorisvandenbossche** [comment]
The `.. currentmodule:: pandas.core.style` in api.rst should probably updated?

**Comment by jreback** [comment]
yep, just pushed it

**Comment by jreback** [comment]
https://github.com/pydata/pandas/commit/8250d7c7ddfd321f91cf7ccd44cb906cdc63c0a0

**Comment by jreback** [comment]
ok, this now built ok for me locally.

Still these but IIRC they were there before
```
None:None: WARNING: toctree contains reference to nonexisting document u'generated/template:'
```

**Pull request closed without merging**